src/w3d/renderer/ddsfile.cpp

```diff
@@ -636,3 +636,48 @@ unsigned DDSFileClass::Calculate_S3TC_Surface_Size(unsigned width, unsigned heig
 
     return ret;
 }
+
+namespace Thyme
+{
+void Change_Texture_File_Extension(char *str, size_t str_len, size_t str_capacity, const char *new_ext)
+{
+    captainslog_dbgassert(str_len < str_capacity, "String length is expected to be smaller than string capacity");
+
+    char *begin = str;
+    char *end = str + str_capacity - 1u;
+    char *ext = str + str_len;
+
+    while (--ext != begin) {
+        char ch = *ext;
+        if (ch == '.') {
+            // File extension found.
+            ext += 1;
+            break;
+        }
+        if (ch == ':' || ch == '/' || ch == '\\') {
+            // No file extension found.
+            ext = begin + str_len;
+            if (ext != end) {
+                *ext++ = '.';
+            }
+            break;
+        }
+    }
+
+    if (ext == begin) {
+        // No file extension found.
+        ext = begin + str_len;
+        if (ext != end) {
+            *ext++ = '.';
+        }
+    }
+
+    while (ext != end && *new_ext != '\0') {
```

Comment:
This can be Replaced with strcat if you null terminate after the '.'

Reply:
That would be no improvement, because then string needs to be walked from the beginning again, which is unnecessary. Algorithm already knows where to start here.

Reply:
You don't need to give it the pointer to the start just give it the one you already have. Also as this is just a generic file extension switcher you can drop the Texture from the name. Then it will be easier to find and replace with `fs::path::replace_extension`.

Reply:
`strlcpy` could be used here, by calculating and passing it the remaining storage for the given buffer. I would refrain from doing that however. I already tested this code here and it works. Rewriting it again would require me testing it again for no real benefit in functionality. It would have to do the same thing that it already does here correctly.